doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+
+      ii.  **Path validation**. The device shall verify that the
+           certificate chain has a valid signature path from a root
+           certificate contained within the device, to the certificate
+           presented by the service.
+           `RFC4158 <https://tools.ietf.org/html/rfc4158>`__ describes
+           this is general. The device is permitted to require a more
+           restricted path, provided the server certificate used
+           complies with this restriction.
+
+      iii. **Validity period**. The validity period of all presented
+           certificates shall be checked against the device’s best
+           notion of the current time.
+
+   c. **Client authentication**. The client shall authenticate itself to
+      the service using a secret known only to that particular device.
+      There are several options, and the technique used is generally
+      mandated by the particular service provider being used.
+
+      i.  **TLS client certificates**. The TLS protocol allows the
+          client to present a certificate, and assert its knowledge of
+          the secret described by that certificate. Generally, these
+          certificates will be stored within the service provider. These
+          certificates can be self-signed, or signed by a CA. Since the
+          service provider maintains a list of valid certificates
+          (mapping them to a device identity), having these certificates
+          signed by a CA does not add any additional security, but may
+          be useful in the management of these certificates.
+
+      ii. **Token-based authentication**. It is also possible for the
+          client to authenticate itself using the *password* field of
+          the MQTT CONNECT packet. However, the secret itself must not
+          be transmitted in this packet. Instead, a token-based
+          protocol, such as
+          `RFC7519 <https://tools.ietf.org/html/rfc7519>`__\ ’s JSON Web
+          Token (JWT) can be used. These tokens will generally have a
+          small validity period (e.g. 1 hour), to prevent them from
+          being reused if they are intercepted. The token shall not be
+          sent until the device has verified the identity of the server.
+
+   d. **Random/Entropy source**. Cryptograph communication requires the
+      generation of secure pseudorandom numbers. The device shall use a
+      modern, accepted cryptographic random-bit generator to generate
+      these random numbers. It shall use either a Non-Deterministic
+      Random Bit Generator (True RGB) implemented in hardware within the
+      SoC, or a Deterministic Random Big Generator (Pseudo RGB) seeded
+      by an entropy source within the SoC.
+
+4. **Communication with the time service**. Ideally, the device shall
+   contain hardware that maintains a secure time. However, most SoCs in
+   use do not have support for this, and it will be necessary to consult
```

Comment:
What are the required characteristics of hardware that supports secure time? Could it just be standard timer peripheral properly "secured", i.e. software-protected? Or are you referring to some specific secure time functionality?

This is also related to the statement

> most SoCs in use do not have support for this

We might need to be a bit more specific to what exact secure time functionality most SOCs do not support...

Reply:
The SOCs that do have clocks generally have a clock that can run based on a separate power connector to the device. There is often support for managing drift, but this would require tracking an external time source and adjusting that drift.

Currently, there seem to be very few RTC drivers in Zephyr at all.